src/Compilers/CSharp/Test/Semantic/Semantics/NullCheckedParameterTests.cs

```diff
@@ -649,21 +649,57 @@ class C
-                                                                           .GetMember<SourceMethodSymbol>("M")
-                                                                           .GetNonNullSyntaxNode()
-                                                                           .DescendantNodes()
-                                                                           .OfType<ParenthesizedLambdaExpressionSyntax>()
-                                                                           .Single();
-            var methodSymbol = (IMethodSymbol)model.GetSymbolInfo(node).Symbol!;
+            ParenthesizedLambdaExpressionSyntax[] nodes = comp.GlobalNamespace.GetTypeMember("C")
+                .GetMember<SourceMethodSymbol>("M")
+                .GetNonNullSyntaxNode()
+                .DescendantNodes()
+                .OfType<ParenthesizedLambdaExpressionSyntax>()
+                .ToArray();
+
+            Assert.Equal(2, nodes.Length);
+
+            var methodSymbol = (IMethodSymbol)model.GetSymbolInfo(nodes[0]).Symbol!;
             Assert.True(methodSymbol.Parameters[0].IsNullChecked);
+
+            methodSymbol = (IMethodSymbol)model.GetSymbolInfo(nodes[1]).Symbol!;
+            Assert.True(methodSymbol.Parameters[0].IsNullChecked);
+            Assert.False(methodSymbol.Parameters[1].IsNullChecked);
+        }
+
+        [Fact]
+        public void NullCheckedDiscard_2()
+        {
+            var source = @"
+using System;
+class C
+{
+    public Action<string, string> action0 = (_, _) => { }; // 1
+    public Action<string, string> action1 = (_!!, _) => { }; // 1
+    public Action<string, string> action2 = (_, _!!) => { }; // 2
+    public Action<string, string> action3 = (_!!, _!!) => { }; // 3, 4
+}";
+            var compilation = CreateCompilation(source, parseOptions: TestOptions.RegularPreview);
+            compilation.VerifyDiagnostics(
+                // (6,46): error CS8990: Discard parameter cannot be null-checked.
+                //     public Action<string, string> action1 = (_!!, _) => { }; // 1
+                Diagnostic(ErrorCode.ERR_DiscardCannotBeNullChecked, "_").WithLocation(6, 46),
+                // (7,49): error CS8990: Discard parameter cannot be null-checked.
+                //     public Action<string, string> action2 = (_, _!!) => { }; // 2
+                Diagnostic(ErrorCode.ERR_DiscardCannotBeNullChecked, "_").WithLocation(7, 49),
+                // (8,46): error CS8990: Discard parameter cannot be null-checked.
+                //     public Action<string, string> action3 = (_!!, _!!) => { }; // 3, 4
+                Diagnostic(ErrorCode.ERR_DiscardCannotBeNullChecked, "_").WithLocation(8, 46),
+                // (8,51): error CS8990: Discard parameter cannot be null-checked.
+                //     public Action<string, string> action3 = (_!!, _!!) => { }; // 3, 4
+                Diagnostic(ErrorCode.ERR_DiscardCannotBeNullChecked, "_").WithLocation(8, 51));
         }
```

Comment:
>} [](http://example.com/codeflow?start=8&length=1)

Consider checking `IParameterSymbol.IsNullChecked` for one or more of these cases.